treeherder/perf/migrations/0013_add_alert_timestamps.py

```diff
@@ -33,11 +33,6 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='performancealertsummary',
             name='last_updated',
-            field=models.DateTimeField(auto_now=True),
-        ),
-        migrations.AlterField(
-            model_name='performancealertsummary',
-            name='created',
-            field=models.DateTimeField(auto_now_add=True, db_index=True),
+            field=models.DateTimeField(null=True),
```

Comment:
I think you could avoid having to override `save` in the model by treating this as two separate migrations: the first with the add fields as they are here and then adding `AlterField` migrations to change that existing field to the `auto_now_add` fields (just be sure its in the proper sequence).

Reply:
Hmm. I don't think I explained myself too well. `auto_now` & `auto_now_add` fill in the new fields with the timestamp when the migration was triggered. That's something me and Dave Hunt want. That's not the real time when an alert was created. We would rather like the **created**/**last_updated** fields to pre-existing `PerformanceAlertSummary` & `PerformanceAlert` be nulled.

Reply:
It looks like you're missing the `AlterField` for `last_updated` on line 33 and also some fields for `AlertSummary`? Can you organize this by model, field name and field changes for each change so it's easier to read at a glance? 

Ed suggested in the channel that it might be a good idea to run the migrations again locally with `./manage sqlmigrate` to see the SQL output to be sure it's doing what is expected (before testing on prototype).

Do you need the prototype's db reset to a previous snapshot or were you able to unapply migrations from your previous trial run?

Reply:
> It looks like you're missing the `AlterField` for `last_updated` on line 33

Line 33 contains does contain an `AlterField` operation. It's adding in the `auto_now` feature for the `last_updated` field. But I do agree it's tricky to read, as `AddField` and `AlterField` are too close to one another.

> and also some field changes for `AlertSummary`?

I think you refer to `AlertSummary`'s `created` field. That field doesn't require any other adding or altering operation. Initially, the `created` field only had the `db_index=True` flag *(and not the `null=True` like `Alert` has)*. I only added `auto_now_add=True` to that, which reflects in this altering operation.

> Can you organize this consistently by model, field name and field changes for each change so it's easier to read at a glance?

I think it's already organized this way. Or you mean to move down the `AlterField` operations under the `AddField` ones? That would be easier on the eyes.

> Ed suggested in the channel that it might be a good idea to run the migrations again locally with `./manage sqlmigrate` to see the SQL output to be sure it's doing what is expected (before testing on prototype).

That's a good idea. This is the SQL equivalent produced.

```sql
BEGIN;
--
-- Add field created to performancealert
--
ALTER TABLE `performance_alert` ADD COLUMN `created` datetime(6) NULL;
--
-- Alter field created on performancealert
--
--
-- Add field first_triaged to performancealert
--
ALTER TABLE `performance_alert` ADD COLUMN `first_triaged` datetime(6) NULL;
--
-- Add field last_updated to performancealert
--
ALTER TABLE `performance_alert` ADD COLUMN `last_updated` datetime(6) NULL;
--
-- Alter field last_updated on performancealert
--
--
-- Add field first_triaged to performancealertsummary
--
ALTER TABLE `performance_alert_summary` ADD COLUMN `first_triaged` datetime(6) NULL;
--
-- Add field last_updated to performancealertsummary
--
ALTER TABLE `performance_alert_summary` ADD COLUMN `last_updated` datetime(6) NULL;
--
-- Alter field last_updated on performancealertsummary
--
--
-- Alter field created on performancealertsummary
--
COMMIT;
```

> Do you need the prototype's db reset to a previous snapshot or were you able to unapply migrations from your previous trial run?

I didn't try this on treeherder-prototype yet. But on previous trials I managed to migrate back.